packages/web/components/Dashboard/Seed.tsx

```diff
@@ -1,36 +1,303 @@
+/* eslint-disable @typescript-eslint/naming-convention */
 import {
+  Box,
+  IconButton,
+  Link,
   Stat,
   StatArrow,
   StatGroup,
   StatHelpText,
   StatLabel,
   StatNumber,
 } from '@metafam/ds';
-import React from 'react';
-
-export const Seed: React.FC = () => (
-  <StatGroup mt={5}>
-    <Stat mb={3}>
-      <StatLabel>Market Price</StatLabel>
-      <StatNumber>$30.77</StatNumber>
-      <StatHelpText>
-        <StatArrow type="increase" />
-        11.43%
-      </StatHelpText>
-    </Stat>
-
-    <Stat mb={3}>
-      <StatLabel>24h Trading Volume</StatLabel>
-      <StatNumber>$1,034</StatNumber>
-      <StatHelpText>
-        <StatArrow type="increase" />
-        9.05%
-      </StatHelpText>
-    </Stat>
-
-    <Stat alignSelf="flex-start" flex="0 0 100%">
-      <StatLabel>24h Low / High</StatLabel>
-      <StatNumber>$30.24 / $32.17</StatNumber>
-    </Stat>
-  </StatGroup>
-);
+import React, { FC, ReactNode, useEffect, useState } from 'react';
+import { FaChartBar } from 'react-icons/fa';
+import {
+  AreaSeries,
+  AreaSeriesPoint,
+  FlexibleXYPlot,
+  GradientDefs,
+  LineSeries,
+} from 'react-vis';
+
+import {
+  findHighLowPrice,
+  HighLow7dType,
+  priceIncreased,
+  ticker,
+  volIncreased,
+  volumeChange,
+} from '../../utils/dashboardHelpers';
+import { apiUrl, chartQuery, tokenId, tokenQuery } from './config';
+
+export type TokenProps = {
+  market_data: MarketDataProps;
+  tickers: Array<TickerProps>;
+};
+export type TokenDataProps = {
+  market: MarketDataProps;
+  ticker: TickerProps;
+  priceUp: boolean;
+  volumeUp: boolean;
+  volumePercent: string;
+  highLow7d: HighLow7dType;
+  prices: Array<Array<number>>;
+};
+
+export type ChartProps = {
+  prices: Array<Array<number>>;
+  total_volumes: Array<Array<number>>;
+};
+
+type MarketDataProps = {
+  price_change_percentage_24h: number | null;
+  total_volume: Record<string, number>;
+  current_price: Record<string, number>;
+};
+
+type TickerProps = {
+  market: MarketProps;
+  token_info_url: string;
+};
+
+type MarketProps = {
+  identifier: string;
+};
+
+export const Seed: FC = () => {
+  const [token, setToken] = useState<TokenDataProps | null>();
+
+  useEffect(() => {
+    if (typeof token !== 'undefined') return;
+    (async () => {
+      try {
+        const tokenResponse = await fetch(
+          `${apiUrl}coins/${tokenId + tokenQuery}`,
+        );
+        const tokenJson = await tokenResponse.json();
+        const chartResponse = await fetch(
+          `${apiUrl}coins/${tokenId + chartQuery}`,
+        );
+        const chartJson = await chartResponse.json();
+
+        const { market_data, tickers } = tokenJson;
+        const { prices, total_volumes } = chartJson;
+
+        const buildTokenData = () => {
```

Comment:
you can just take everything out of the function and pass the json object to setToken